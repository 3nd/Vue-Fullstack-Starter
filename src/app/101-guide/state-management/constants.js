// @flow

export interface ICounter {
  value: number;
  step: number;
  stepList: number[];
}

export interface IContent {
  state: ICounter;
  commit(mutation: string, payload?: any): void;
  dispatch(action: string): void;
}

export const INITIAL: ICounter = {
  value: 0,
  step: 1,
  stepList: [...Array(10).fill(1).map((x, y) => x + y)],
};
