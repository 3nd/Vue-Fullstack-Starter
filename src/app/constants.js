// @flow

import { IApp } from './types';

export const INITIAL: IApp = {
  theme: localStorage.getItem('theme') || 'light',
  navigation: [
    // standard
    { icon: 'face', text: 'helloWorld', route: '/hello-world', subheader: 'standard' },
    {
      icon: 'loyalty',
      text: '101Guide',
      children: [
        {
          text: 'components',
          children: [
            { text: 'moduleSystems', route: '/foo', disabled: true },
            { text: 'props', route: '/101-guide/components/props' },
            { text: 'slots', route: '/101-guide/components/slots' },
          ],
        },
        {
          text: 'transitions',
          children: [
            { text: 'state', route: '/foo', disabled: true },
          ],
        },
        { text: 'directives', route: '/101-guide/directives' },
        { text: 'filters', route: '/101-guide/filters' },
        {
          text: 'routing',
          children: [
            { text: 'params', route: '/foo', disabled: true },
            { text: 'guards', route: '/foo', disabled: true },
          ],
        },
        {
          text: 'stateManagement',
          children: [
            { text: 'modules', route: '/foo', disabled: true },
          ],
        },
      ],
    },
    {
      icon: 'business',
      text: 'crudOperations',
      children: [
        { text: 'basic', route: '/crud-operations/basic' },
        { text: 'rest', route: '/crud-operations/rest' },
        { text: 'graphql', route: '/crud-operations/graphql' },
      ],
    },
    {
      icon: 'contacts',
      text: 'formControls',
      children: [
        { text: 'textFields', route: '/form-controls/text-fields' },
        { text: 'selects', route: '/form-controls/selects' },
        { text: 'selectionControls', route: '/form-controls/selection-controls' },
        { text: 'pickers', route: '/form-controls/pickers' },
      ],
    },
    {
      icon: 'fa fa-table',
      text: 'dataTable',
      children: [
        { text: 'basic', route: '/data-table/basic' },
        { text: 'rest', disabled: true },
        { text: 'graphql', disabled: true },
      ],
    },
    { icon: 'language', text: 'internationalization', route: '/internationalization' },
    { icon: 'fa fa-bar-chart', text: 'interactiveChart', route: '/interactive-chart' },
    {
      icon: 'verified_user',
      text: 'authorization',
      disabled: true,
      children: [
        { text: 'rest', disabled: true },
        { text: 'graphql', disabled: true },
      ],
    },
    { icon: 'fa fa-credit-card', text: 'payment', disabled: true },
    {
      icon: 'message',
      text: 'messages',
      disabled: true,
      children: [],
    },

    // playground
    { icon: 'fa fa-calculator', text: 'counter', route: '/counter', subheader: 'playground' },
    { icon: 'photo_library', text: 'gallery', route: '/gallery', disabled: true },
    { icon: 'fa fa-qrcode', text: 'qrCode', route: '/qrcode', disabled: true },
    { icon: 'store', text: 'store', route: '/store', disabled: true },

    // profound
    { icon: 'av_timer', text: 'realTime', route: '/real-time', disabled: true, subheader: 'profound' },
  ],
  drawer: false || null,
  languages: [
    { key: 'en', label: 'English' },
    { key: 'zh', label: '中文' },
    { key: 'ja', label: '日本語' },
  ],
};
