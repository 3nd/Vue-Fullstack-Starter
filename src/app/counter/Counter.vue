<template>
  <div>
    <div class="text">
      <span class="md-title">
        Clicked: {{ $store.state.counter.value }} times, count is {{ evenOrOdd }}.
      </span>
    </div>

    <div>
      <md-button class="md-raised md-primary" @click="increment">Increment</md-button>
      <md-button class="md-raised md-primary" @click="decrement">Decrement</md-button>
      <md-button class="md-raised md-accent" @click="incrementIfOdd">Increment if odd</md-button>
      <md-button class="md-raised md-accent" @click="incrementAsync">Increment async</md-button>
    </div>
  </div>
</template>

<script>
import { mapGetters, mapActions } from 'vuex';

export default {
  // name: 'counter',
  computed: mapGetters([
    'evenOrOdd'
  ]),
  methods: mapActions([
    'increment',
    'decrement',
    'incrementIfOdd',
    'incrementAsync'
  ])
};
</script>

<style scoped>
.text {
  padding: 1rem;
}
</style>
