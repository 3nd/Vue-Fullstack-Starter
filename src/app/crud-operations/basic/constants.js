export const origin = [
  { id: 1, primary: 'Vanilla', accent: 'MobX' },
  { id: 2, primary: 'Angular', accent: 'NgRx' },
  { id: 3, primary: 'React', accent: 'Redux' },
  { id: 4, primary: 'Vue', accent: 'Vuex' },
];

export const INITIAL = {
  headers: [
    { text: 'ID', value: 'id' },
    { text: 'Primary', value: 'primary' },
    { text: 'Accent', value: 'accent' },
    { text: 'Actions', value: 'actions' },
  ],

  dataset: [...origin],

  addData: { primary: '', accent: '' },
  searchData: { primary: '', accent: '' },
};
