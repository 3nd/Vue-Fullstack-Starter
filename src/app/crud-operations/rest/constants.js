export const API_LIST = `${process.env.API_URL}/__/list`;

export const INITIAL = {
  dataset: [],
  addData: { text: '' },
  searchData: { text: '' },
  editData: { _id: 0, text: '', dialog: false },
  deleteData: { _id: 0, dialog: false },
  loading: false,
};

export const SUCCESS = '[REST] SUCCESS';
export const FAILURE = '[REST] FAILURE';
