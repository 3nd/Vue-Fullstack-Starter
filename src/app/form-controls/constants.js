export const INITIAL = {
  // input
  nickname: '',

  // Input validation

  // select

  // multiple select

  // nested select

  // checkboxes

  // radio buttons
  gender: '',

  // switch

  // slider

  // autocomplete

  // datepicker

  // datepicker with range

  // timepicker

  // colorpicker

  // file upload
};
