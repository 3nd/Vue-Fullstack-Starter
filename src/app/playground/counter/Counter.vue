<script>
import { mapActions, mapGetters } from 'vuex';

export default {
  methods: mapActions([
    'increment',
    'decrement',
    'incrementAsync',
    'decrementAsync',
    'incrementIfOdd',
    'decrementIfEven'
  ]),
  computed: {
    $c() {
      return this.$store.state.counter;
    },
    ...mapGetters(['evenOrOdd'])
  },
  render() {
    return (
      <div>
        <div class="title">
          <p class="md-title">
            Clicked: <span class="title--highlight">{this.$c.value}</span> times, value is <span class="title--highlight">{this.evenOrOdd}</span>.
          </p>
        </div>

        <div>
          <md-button class="md-raised md-primary" onClick={this.increment}>Increment</md-button>
          <md-button class="md-raised md-primary" onClick={this.decrement}>Decrement</md-button>
        </div>

        <div>
          <md-button class="md-raised md-primary" onClick={this.incrementAsync}>Increment (Async)</md-button>
          <md-button class="md-raised md-primary" onClick={this.decrementAsync}>Decrement (Async)</md-button>
          <md-button class="md-raised md-primary" onClick={this.incrementIfOdd}>Increment (If Odd)</md-button>
          <md-button class="md-raised md-primary" onClick={this.decrementIfEven}>Decrement (If Even)</md-button>
        </div>

        <div>
          <md-button class="md-raised md-primary">Start Count</md-button>
          <md-button class="md-raised md-primary">Cancel Count</md-button>
        </div>
      </div>
    );
  }
};
</script>

<style scoped>
.title {
  padding: 0 .5rem;

  /* BUG: vue-jest, https://github.com/vuejs/vue-jest/issues/51 */
  /* &--highlight {
    font-weight: bold;
  } */
}
</style>
