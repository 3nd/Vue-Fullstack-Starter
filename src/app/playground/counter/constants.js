// @flow

export interface ICounter {
  value: number;
}

export interface IContent {
  commit(mutation: string): void;
  state: ICounter;
}

export const INITIAL: ICounter = {
  value: 0,
};
