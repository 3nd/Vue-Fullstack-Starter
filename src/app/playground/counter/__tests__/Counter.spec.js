import Vuex from 'vuex';
import { shallow, createLocalVue } from 'vue-test-utils';

import { INITIAL, INCREMENT, DECREMENT } from '../constants';
import Counter from '../Counter';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Counter', () => {
  let [wrapper, store] = [];
  let [state, actions, mutations, getters] = [];

  beforeEach(() => {
    state = INITIAL;

    actions = {
      increment: jest.fn(),
      decrement: jest.fn(),
      incrementAsync: jest.fn(),
      decrementAsync: jest.fn(),
      incrementIfOdd: jest.fn(),
      decrementIfEven: jest.fn()
    };

    mutations = {
      [INCREMENT]: jest.fn(),
      [DECREMENT]: jest.fn()
    };

    getters = {
      evenOrOdd: jest.fn(() => 'even')
    };

    store = new Vuex.Store({
      modules: {
        counter: {
          state,
          actions,
          mutations,
          getters
        }
      }
    });

    wrapper = shallow(Counter, { store, localVue });
  });

  it('should render initial component', () => {
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
