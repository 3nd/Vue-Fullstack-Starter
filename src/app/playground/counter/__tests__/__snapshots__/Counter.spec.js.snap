// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render initial component 1`] = `
<div>
  <div
    class="title"
  >
    Clicked: 
    <span
      class="title--highlight"
    >
      0
    </span>
     times, value is 
    <span
      class="title--highlight"
    >
      even
    </span>
    .
  </div>
  <div>
    <v-btn
      color="info"
    >
      Increment
    </v-btn>
    <v-btn
      color="info"
    >
      Decrement
    </v-btn>
  </div>
  <div>
    <v-btn
      color="info"
    >
      Increment (Async)
    </v-btn>
    <v-btn
      color="info"
    >
      Decrement (Async)
    </v-btn>
    <v-btn
      color="info"
    >
      Increment (If Odd)
    </v-btn>
    <v-btn
      color="info"
    >
      Decrement (If Even)
    </v-btn>
  </div>
  <div>
    <v-btn
      color="info"
    >
      Start Count
    </v-btn>
    <v-btn
      color="info"
    >
      Cancel Count
    </v-btn>
  </div>
</div>
`;
